### ISSUE TITLE
Add recipie for aluminum arrows and bolts

### ISSUE THREAD

**Issue by Proxiehunter** [cataclysmbnteam/Cataclysm-BN#5667]
### Is your feature request related to a problem? Please describe.

[Aluminum is readily available](https://cbn-guide.mythoscraft.org/item/material_aluminium_ingot). We _have_ the ability to forge things in game. [Aluminum arrows](https://cbn-guide.mythoscraft.org/item/arrow_metal) and [bolts](https://cbn-guide.mythoscraft.org/item/bolt_steel) exist in game in all the same forms _wooden_ bolts are found.

We can even hook up a 3D printer and make _carbon fiber_ [arrows](https://cbn-guide.mythoscraft.org/item/arrow_cf) and [bolts](https://cbn-guide.mythoscraft.org/item/bolt_cf).  The making of aluminum arrows and bolts doesn't seem like it should be beyond a characters abilities. And looking at the json for the broadhead version of both wooden and aluminum arrows it looks like the aluminum versions the player _can't_ make are superior to the wooden versions the player _can_ because the aluminum arrows have lower dispersion (assuming I'm right about _lower_ dispersion being good) and a higher chance of being recovered after being fired. 

### Describe the solution you'd like

Would like to see recipes added for the player to make the aluminum versions of broadhead, field point, and small game arrows and bolts. Should be relatively easy json work if the required ingredients and tools can be agreed on. 

While we're at it consider if maybe compound bows and crossbows should _also_ be craftable.

**Comment by blot-runner** [comment]
> While we're at it consider if maybe compound bows and crossbows should _also_ be craftable.
What do you mean, 'should be'? Aren't they already?
![image](https://github.com/user-attachments/assets/9a774511-7d54-4f03-96eb-3356006c3a2c)

**Comment by 9SineD** [comment]
I think that aluminum rods can be obtained by sawing larger ingots or casting it into a rough shape. Then turn the rods in a cordless drill, like on an improvised lathe, fortunately aluminum is quite soft. Of course, we don't have a normal file, so we would have to come up with something.

**Comment by Proxiehunter** [comment]
> > While we're at it consider if maybe compound bows and crossbows should _also_ be craftable.
> 
> What do you mean, 'should be'? Aren't they already? 

I think you might have a mod that enables them to be crafted. There's no recipe listed on the Hitchhikers Guide to the Cataclysm and just to be sure I searched both Compound and Bow in my crafting menu, hitting u to see unavailable recipes and compound bows are not in my crafting menu (and I am using the November 3rd 2024 nightly build). I suppose it's possible that one of my mods, either through error or intentionally, _removes_ the ability to craft them but you running a mod that _adds_ them being craftable seems more likely. 

It _does_ say in your screen shot the origin is Bright Nights, but the bows do _exist_ in game just not as far as I can tell as something the player can make on their own without a mod.

**Comment by ZChris13** [comment]
I also do not have compound bows in my crafting list

**Comment by ZChris13** [comment]
shower thought: it might be better to have a bunch of aluminum arrow shafts spawn in sporting goods stores and megastores, and then be able to craft heads onto them
with their decent chance of remaining intact once fired, you would ideally only need to visit a single sporting goods store in order to be set for a while
this would also require disassembly info for the three types of aluminum arrows

**Comment by blot-runner** [comment]
> I think that aluminum rods can be obtained by sawing larger ingots or casting it into a rough shape. Then turn the rods in a cordless drill, like on an improvised lathe, fortunately aluminum is quite soft. Of course, we don't have a normal file, so we would have to come up with something.

But do we need such complexity? It's not that of a high-end item to justify an involved crafting process. And many steel items are already crafted from ingots/chunks/scrap (provided you have an "anvil-hammer-chisel-forge-thongs-crucible-swage&die" combo), even though it's probably not that simple a process irl.
IDK, I would rather have something like this (see a screenshot - hacked up a quick recipe, can share the file if smb wants to implement it).
![image](https://github.com/user-attachments/assets/0ed01d27-3970-437c-9797-cf304dee9fcc)

**Comment by blot-runner** [comment]
> shower thought: it might be better to have a bunch of aluminum arrow shafts spawn in sporting goods stores and megastores, and then be able to craft heads onto them with their decent chance of remaining intact once fired, you would ideally only need to visit a single sporting goods store in order to be set for a while this would also require disassembly info for the three types of aluminum arrows

Maybe, but IMO, the player should be able to craft those, too.
I mean, we can forge a KATANA from scrap metal, can't we? Sure some aluminun rods aren't that hard.

**Comment by 9SineD** [comment]
Of course your option would work too, in fact most of the crafting requirements are based on the tip, not the shaft. In my opinion simple electrical processing would justify the increased accuracy of such projectiles, due to greater standardization.

IRL I would not want to forge each shaft separately, on the other hand you forge the tips anyway. :(

**Comment by blot-runner** [comment]
> Of course your option would work too, in fact most of the crafting requirements are based on the tip, not the shaft. In my opinion simple electrical processing would justify the increased accuracy of such projectiles, due to greater standardization.
Eh, can be argued either way. Some of the projects the character undertakes call for a much higher precision than simple arrows making. Of course, one can instead add craftable "makeshift aluminum" arrows and bolts that'd be as durable, but less precise than the pre-Cataclysm ones, but I, personally, would rather not inflate the items list if I can help it.

> IRL I would not want to forge each shaft separately, on the other hand you forge the tips anyway. :(
Well, the way it played in my head was you heat the metal until it's soft enough, then draw it through the swages, sorta like the wire was made in the past. Then you let the rod cool off and saw it into the actual shafts. I admit using beer cans or foil for that seems less plausible, but, aluminum having lower melting point than steel, I assume one can cast it into the ingot using the crucible.
But I don't know first thing about metalworking, so take it all with a grain of salt. ;-)

**Comment by blot-runner** [comment]
> A molding process doesn't _have_ to be much more involved. Although if a molding method is used then creating a new "casting mold" tool might be best. And then you don't need the metal sawing. Would kind of suck to make a tool for only one recipe though. On the other hand the sand in a sand casting mold is reusable and can be reshaped into whatever pattern you like which could lead to recipes for "molded" versions of various tools and maybe some weapons (_other_ than aluminum arrows and bolts) that are basically a _faster_ way to make the same tool/weapon than forging them at your anvil.
> 
> Mostly I'm thinking basic tool box tools and knives. You could probably sand cast parts of some medieval weaponry though. Put the crafting of the mold at about Fab 5 and you're not able to make the faster versions until after you're able to make most tool box tools and besides the mold is extra resources and time to craft.
> 
> Turning them on a drill seems excessive but as I said above adding some more sand to represent using it to scour the flash off the shafts wouldn't be a bad idea.

Hmm, but do we need a dedicated mold item for that? There's already a forge, a crucible, sand and planks as tools plus a bit of sand as a component. The mold itself can be abstracted out, it's rebuit every time anyway.

**Comment by ZChris13** [comment]
> Maybe, but IMO, the player should be able to craft those, too. I mean, we can forge a KATANA from scrap metal, can't we? Sure some aluminun rods aren't that hard.
a Katana is significantly easier to forge than straight and true aluminum shafts
there have been significant developments recently in the 3D printed gun space regarding using drills and jigs to function as a primitive mill, I recommend including "drilling" tools in the aluminum arrow recipe as well as the standard anvil/forge metalworking stuff with swage and die set

**Comment by blot-runner** [comment]
> a Katana is significantly easier to forge than straight and true aluminum shafts

I'm tempted to ask my esteemed opponent to quote their source of that insight) Last time I checked, a katana should've also been straight, at least in one projection. One can also forge fencing foils, if that's a better approximation for the arrow shaft. To say nothing about ability to fabricate seemingly factory-grade suppressors, caseless ammo, bionic modules... I think, it kinda gives us a ballpark estimate of the character's crafting ability, and if a straight rod seems harder to make than those, well...)


> there have been significant developments recently in the 3D printed gun space regarding using drills and jigs to function as a primitive mill, I recommend including "drilling" tools in the aluminum arrow recipe as well as the standard anvil/forge metalworking stuff with swage and die set to represent truing the drawn aluminum with a primitive lathe

IDK, I'd rather have it as a separate recipe, something requiring Fabrication 4-5, a cordless drill (or a lathe when one is implemented, I'm sure by now it's not a question of "if") and Filing quality (finally, something to use that metal fileset for). That way, a player will be able to choose between needing a higher skill (but being able to craft all your tools) and having to find a somewhat uncommon item.

**Comment by RoyalFox2140** [comment]
Most of the ideas and discussion here is generally too complicated. We're not really going to model the difference between forging and shaping steel versus aluminum. When aluminum arrows get added It will likely be the same as forging steel. It's not worthwhile to do a complicated setup for a narrow number of niche items.

**Comment by ZChris13** [comment]
> Most of the ideas and discussion here is generally too complicated. We're not really going to model the difference between forging and shaping steel versus aluminum. When aluminum arrows get added It will likely be the same as forging steel. It's not worthwhile to do a complicated setup for a narrow number of niche items.

yes, but the steel bolts that exist are fluffed and statted as primitive and crude, may I recommend including simply the drilling quality on top of the normal metalworking blueprint to make it more difficult?
